### PR TITLE
global: migrate CI to GH Actions

### LINKED ISSUE #235: global: migration to github-actions from travis

### PR DESCRIPTION
**Pull request by diegodelemos**
* Closes #235.

Co-authored-by: ParthS007 <parth.shandilya@cern.ch>

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+            EXTRAS: "all,mysql"
+
+          - db-service: mysql8
+            DB: mysql
+            MYSQL_VERSION: MYSQL_8_LATEST
+            SQLALCHEMY_DATABASE_URI: "mysql+pymysql://invenio:invenio@localhost:3306/invenio"
+            EXTRAS: "all,mysql"
+
+          - python-requirements: dev
+            experimental: true
```
Comment (diegodelemos) [inline .github/workflows/tests.yml:62]: Emulates allowed failures from Travis: see docs [here](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#example-preventing-a-specific-failing-matrix-job-from-failing-a-workflow-run)

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+          db-service: [postgresql9, postgresql11, mysql5, mysql8]
+          exclude:
+          - python-version: 3.8
+            python-requirements: min
+
+          - db-service: postgresql11
+            python-version: 3.6
+
+          - db-service: mysql8
+            python-version: 3.6
```
Comment (diegodelemos) [inline .github/workflows/tests.yml:34]: @lnielsen adding exceptions should be as simple as these examples.
- Are these correct?
- Do you have any others in mind?

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -0,0 +1,109 @@
+name: CI
+
+on:
+  push:
+    branches: gh-actions
+  pull_request:
+    branches: gh-actions
```
Comment (mvidalgarcia) [inline .github/workflows/tests.yml:7]: Minor: Just a reminder that this has to be `master`

File: .github/workflows/tests.yml (hunk 1 above)
Comment (mvidalgarcia) [reply .github/workflows/tests.yml:62]: Minor: Maybe we can add a comment to explain it?

Something like: "If `python-requirements` is `dev`, the GitHub workflow won't fail."

File: .github/workflows/tests.yml [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+            SQLALCHEMY_DATABASE_URI: "mysql+pymysql://invenio:invenio@localhost:3306/invenio"
+            EXTRAS: "all,mysql"
+
+          - db-service: mysql8
+            DB: mysql
+            MYSQL_VERSION: MYSQL_8_LATEST
+            SQLALCHEMY_DATABASE_URI: "mysql+pymysql://invenio:invenio@localhost:3306/invenio"
+            EXTRAS: "all,mysql"
+
+          - python-requirements: dev
```
Comment (lnielsen) [inline .github/workflows/tests.yml:61]: Comment:  I think these we can simply remove. We originally had dev requirements to be warned early, but honestly it's really rarely that it has actually been helpful.....

File: .github/workflows/tests.yml [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+        required: false
+        default: 'Manual trigger'
+
+jobs:
+  Tests:
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+          python-version: [3.6, 3.7, 3.8]
+          python-requirements: [min, pypi, dev]
```
Comment (mvidalgarcia) [inline .github/workflows/tests.yml:24]: Minor: What do you think about calling it `requirements-level` instead?

**Review by lnielsen — APPROVED** [review]

File: .github/workflows/tests.yml (hunk 4 above)
Comment (mvidalgarcia) [reply .github/workflows/tests.yml]: We can get rid of the ugly "Generate dependencies devel" step then 🚀

**Pull request merged**